speechbrain/processing/features.py

```diff
@@ -1153,6 +1154,15 @@ def forward(self, x, lengths, spk_ids=torch.tensor([]), epoch=0):
             current_mean = torch.mean(torch.stack(current_means), dim=0)
             current_std = torch.mean(torch.stack(current_stds), dim=0)
 
+            # If multi GPU, we need to sync the statistics across processes.
+            if is_distributed_initialized():
+                torch.distributed.all_reduce(
+                    current_mean, op=torch.distributed.ReduceOp.AVG
+                )
+                torch.distributed.all_reduce(
+                    current_std, op=torch.distributed.ReduceOp.AVG
+                )
+
```

Comment:
I'm wondering if we should add something to the distributed tools to ensure this won't hang if its in a `if_main_process` block, similar to this broadcast code:

https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/utils/distributed.py#L205

Reply:
Oh right, good catch.

Reply:
Done!